src/hooks/useUnreadMessages.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState, createContext, useContext, useEffect } from 'react';
+import { onNotificationReceived } from '../common/Notifications';
+import { uniq } from 'lodash';
+import { useAsyncStorage } from './useAsyncStorage';
+
+type UnreadMessageContextProps = {
+  unreadMessagesUserIds?: string[];
+  markMessageRead?: (userId: string) => void;
+};
+
+const UnreadMessagesContext = createContext<UnreadMessageContextProps>({});
+
+export const UnreadMessagesContextProvider = ({
+  children,
+}: {
+  children?: React.ReactNode;
+}) => {
+  const [initialUnreadMessages, setStoredUnreadMessages] = useAsyncStorage(
+    'UnreadMessageUserIds',
+  );
+
+  const [unreadMessagesUserIds, setUnreadMessages] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (initialUnreadMessages.isFetchedAfterMount) {
+      setUnreadMessages(
+        initialUnreadMessages.data
+          ? initialUnreadMessages.data?.split('#')
+          : [],
+      );
+    }
+  }, [initialUnreadMessages.data, initialUnreadMessages.isFetchedAfterMount]);
+
+  onNotificationReceived((notification) => {
+    if (notification.payload.notificationType === 'privatePost') {
+      const incomingUserId = `${notification?.payload?.authorId}`;
+      const computeIds = (unreadIds?: string[]) => {
+        return unreadIds
+          ? uniq([...unreadIds, incomingUserId])
+          : [incomingUserId];
+      };
+      setUnreadMessages((currentUnreads) => {
+        const newUserIds = computeIds(currentUnreads);
+        setStoredUnreadMessages(newUserIds.join('#'));
+        return newUserIds;
+      });
+    }
+  });
```

Comment:
Should this be in a `useEffect` so we only register the handler once?

Reply:
Yes, probably, but this particular code will no longer exists in the latest commit.